### PR TITLE
stacks: update RPC APIs with deferral information

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
This PR adds representation of deferred changes into the stacks RPC API. We create a new kind of planned change (a deferred resource) and emit that whenever we discover deferred changes from Terraform Core.

### CODE REVIEW

File: internal/stacks/stackplan/from_proto.go [hunk 1]
```diff
@@ -204,6 +206,79 @@ func LoadFromProto(msgs []*anypb.Any) (*Plan, error) {
 				c.ResourceInstancePriorState.Put(fullAddr, nil)
 			}
 
+		case *tfstackdata1.PlanDeferredResourceInstanceChange:
```
Comment (alisdair) [inline internal/stacks/stackplan/from_proto.go:209]: There's a lot of overlap between this case and the planned change, which makes sense. Do you think there might be some way to unify them, maybe with helper functions? I'm a bit concerned about logic drift over time if the code is duplicated.

File: internal/stacks/stackplan/planned_change.go [hunk 2]
```diff
@@ -277,18 +277,12 @@ func (pc *PlannedChangeResourceInstancePlanned) PlannedChangeProto() (*terraform
 		// if this plan later gets applied.
 		// We only emit a "raw" in this case, because this is a relatively
 		// uninteresting edge-case.
```
Comment (alisdair) [inline internal/stacks/stackplan/planned_change.go:279]: I'm not sure this comment fits here any more, although I am having some trouble with GitHub's diff of this file…

**Review by alisdair — COMMENTED** [review]
This looks good to me, as far as I understand it!

Are we intentionally not adding deferral data to the JSON change description documents in this PR, or do I misunderstand something?

File: internal/stacks/stackplan/from_proto.go (hunk 1 above)
Comment (liamcervante) [reply internal/stacks/stackplan/from_proto.go:171]: Done!

File: internal/stacks/stackplan/planned_change.go (hunk 2 above)
Comment (liamcervante) [reply internal/stacks/stackplan/planned_change.go]: Moved the comments around so they line up properly again.

**Review by alisdair — APPROVED** [review]

**Pull request merged**